src/JsonSchema/Constraints/TypeConstraint.php

```diff
@@ -48,7 +48,7 @@ public function check($value = null, $schema = null, $path = null, $i = null)
             $validatedOneType = false;
             $errors = array();
             foreach ($type as $tp) {
-                $validator = new static($this->checkMode);
+                $validator = $this->getFactory()->createInstanceFor('type');
```

Comment:
I support this. However, `checkMode` can be set in the constructor (and you probably don't need to use the Factory, even if you should). So I'm not sure, if there was a reason why the constructor with the explicit checkMode was used or not (another contributor probably knows more about that).